### PR TITLE
ISSUE-5472: Now possible to resimulate if failure occurs during initialisation

### PR DESCRIPTION
**Pull request by JKRT**
The reason for making this change was discussed with @casella in https://trac.openmodelica.org/OpenModelica/ticket/5472#comment:6

### CODE REVIEW

File: OMEdit/OMEdit/OMEditGUI/Simulation/SimulationDialog.cpp [hunk 1]
```diff
@@ -2175,44 +2174,45 @@ void SimulationDialog::showArchivedSimulation(QTreeWidgetItem *pTreeWidgetItem)
...
 void SimulationDialog::simulate()
 {
-  if (validate()) {
-    // interactive simulation
-    if (mpInteractiveSimulationGroupBox->isChecked() || mIsReSimulate) {
+  if (!validate()) {
+    return;
+  }
+  // if no option is selected then show error message to user
+  if (!(mpSaveExperimentAnnotationCheckBox->isChecked() ||
```
Comment (JKRT) [inline OMEdit/OMEdit/OMEditGUI/Simulation/SimulationDialog.cpp]: This check was never executed before running interactive simulations, I rewrote the logic to make the nesting not as deep and that we do this check for both simulation possibilities

**Comment by JKRT** [comment]
@adeas31 if it is all ok for you, could you merge?

**Comment by adeas31** [comment]
I think this solution is not the best one. What will happen if the simulation fails and there is no result file?

My suggestion is to add re-simulate buttons to the simulation output window. Enable the buttons after the successful compilation and one simulation run.

**Comment by casella** [comment]
@JKRT, does this fix [#5472](https://trac.openmodelica.org/OpenModelica/ticket/5472)?

**Comment by JKRT** [comment]
> 
> 
> @JKRT, does this fix [#5472](https://trac.openmodelica.org/OpenModelica/ticket/5472)?

It does but as Adeel pointed out not in a nice way (Might take one-two hours two fix in a nice way. Sadly I am drowning in alot of other work:(, I will open the branch again)

**Pull request closed without merging**